src/Control/Carrier/TaskPool.hs

```diff
@@ -13,20 +13,22 @@ import Control.Carrier.Reader
 import Control.Carrier.Threaded
 import Control.Concurrent.STM
 import Prologue
+import Control.Effect.Lift (sendIO)
 
-withTaskPool :: (Has (Lift IO) sig m, MonadIO m)
+withTaskPool :: Has (Lift IO) sig m
```

Comment:
switched all of the uses of `MonadIO`/`liftIO` here to `Has (Lift IO)`/`sendIO`, because MonadIO is redundant with Lift IO